cmd/codegen/generator/generator.go

```diff
@@ -23,6 +23,12 @@ const (
 	SDKLangTypeScript SDKLang = "typescript"
 )
 
+type DependencyConfig struct {
+	Name string
+	Pin  string
+	Ref  string
+}
```

Comment:
I don't think this needs re-declaration actually - you should just be able to use this directly: https://github.com/dagger/dagger/blob/84ee1197610edd94b8997323bc7c7e7f379e288a/core/modules/config.go#L183-L193

It's a separate package so we can easily automatically generate docs from it.